src/auth/baseexternalclient.ts

```diff
@@ -401,9 +408,21 @@ export abstract class BaseExternalAccountClient extends AuthClient {
     };
 
     // Exchange the external credentials for a GCP access token.
-    const stsResponse = await this.stsCredential.exchangeToken(
-      stsCredentialsOptions
-    );
+    let stsResponse;
+    // If there is work force pool user project but no client defined,
+    // pass user project config for STS token exchange.
+    // Otherwise, only pass stsCredentialsOptions for token exchange.
+    if (!this.clientAuth && this.workforcePoolUserProject) {
+      stsResponse = await this.stsCredential.exchangeToken(
+        stsCredentialsOptions,
+        undefined,
+        {user_project: this.workforcePoolUserProject}
+      );
+    } else {
+      stsResponse = await this.stsCredential.exchangeToken(
+        stsCredentialsOptions
+      );
+    }
```

Comment:
Note that this change is specific to IdentityPoolClient.

Reply:
I mentioned in the description. I cant change it here, since baseExternalClient is the parent class of IdentityPoolClient, the STS exchange this.stsCredential.exchangeToken in refreshAccessTokenAsync() is non-public and only implemented in BaseExternalClient.

Reply:
This can definitely be changed, but I'll leave it up to @bojeil-google.

Reply:
I think we should support this in base external account class (as @xil222 is already doing). See other comment for reasoning.